### PR TITLE
fix: use new ArrayBuffer::Allocator::Realloc API

### PR DESCRIPTION
**Pull request by ckerr**
Use the new ArrayBuffer::Allocator::Realloc() method added to libcc `master` in https://github.com/electron/libchromiumcontent/pull/625

cc @codebytere @alexeykuzmin

### CODE REVIEW

File: src/node_buffer.cc [hunk 1]
```diff
@@ -254,8 +254,10 @@ MaybeLocal<Object> New(Isolate* isolate,
       isolate->GetArrayBufferAllocator()->Free(data, length);
       data = nullptr;
     } else if (actual < length) {
-      // We should call realloc here, but v8::ArrayBufferAllocator does not
-      // provide such ability.
+      data = isolate->GetArrayBufferAllocator()->Realloc(data, length);
```
[the remark this answers is not in the document]
Comment (ckerr) [reply src/node_buffer.cc]: Right on both counts. Thanks for the review.

**Comment by ckerr** [comment]
@addaleax Actually if I'm reading this right... do we need an `ArrayBuffer::Allocator::Realloc()` method at all?

I assumed we did due to the previous comment, but this new patch seems like it would effectively do the same thing and would remove the need for an upstream libcc patch to V8 / WebKit / gin...

**Comment by addaleax** [comment]
@ckerr Okay, sorry, I didn’t have the context here and only looked at the diff… Yes, this new patch should do the trick as well.

File: src/node_buffer.cc [hunk 2]
```diff
@@ -254,7 +254,10 @@ MaybeLocal<Object> New(Isolate* isolate,
       isolate->GetArrayBufferAllocator()->Free(data, length);
       data = nullptr;
     } else if (actual < length) {
-      data = isolate->GetArrayBufferAllocator()->Realloc(data, actual);
+      auto long_data = data;
+      data = static_cast<char*>(BufferMalloc(isolate, actual));
```
Comment (addaleax) [inline src/node_buffer.cc]: should this use the ArrayBufferAllocator’s alloc function?

**Comment by addaleax** [comment]
@codebytere Could 6ef4b2a3422d40dad6c96d8e2899136146bec866 and this potentially be upstreamed into Node?

**Comment by codebytere** [comment]
@addaleax will do!

**Pull request merged**